src/secret/ledger_nano.rs

```diff
@@ -101,154 +129,160 @@ impl SecretManageExt for LedgerSecretManager {
-    //                                     if *remainder_address.unwrap() == *e.address() {
-    //                                         break;
-    //                                     }
-    //                                 }
-    //                             }
-    //                         }
-    //                         _ => {
-    //                             log::debug!("[LEDGER] unsupported output");
-    //                             return Err(crate::Error::LedgerMiscError);
-    //                         }
-    //                     }
-    //                     remainder_index += 1;
-    //                 }
-
-    //                 // was index found?
-    //                 if remainder_index as usize == essence.outputs().len() {
-    //                     log::debug!("[LEDGER] remainder_index not found");
-    //                     return Err(crate::Error::LedgerMiscError);
-    //                 }
-    //             }
-    //         }
-    //     }
-
-    //     // pack essence into bytes
-    //     let essence_bytes = essence.pack_to_vec();
-
-    //     // prepare signing
-    //     log::debug!("[LEDGER] prepare signing");
-    //     log::debug!(
-    //         "[LEDGER] {:?} {:?} {} {} {:?}",
-    //         input_bip32_indices,
-    //         essence_bytes,
-    //         has_remainder,
-    //         remainder_index,
-    //         remainder_bip32
-    //     );
-    //     ledger.prepare_signing(
-    //         input_bip32_indices,
-    //         essence_bytes,
-    //         has_remainder,
-    //         remainder_index,
-    //         remainder_bip32,
-    //     )?;
-
-    //     // show essence to user
-    //     // if denied by user, it returns with `DeniedByUser` Error
-    //     log::debug!("[LEDGER] await user confirmation");
-    //     ledger.user_confirm()?;
-
-    //     // sign
-    //     let signature_bytes = ledger.sign(input_len as u16)?;
-    //     let mut readable = &mut &*signature_bytes;
-    //     // unpack signature to unlockblocks
-    //     let mut unlocks = Vec::new();
-    //     for _ in 0..input_len {
-    //         let unlock = Unlock::unpack_verified(&mut readable).map_err(|_| crate::Error::PackableError)?;
-    //         unlocks.push(unlock);
-    //     }
-    //     Ok(unlocks)
-    // }
+        // show essence to user
+        // if denied by user, it returns with `DeniedByUser` Error
+        log::debug!("[LEDGER] await user confirmation");
+        ledger.user_confirm()?;
+
+        // sign
+        let signature_bytes = ledger.sign(input_len as u16)?;
+        let mut readable = &mut &*signature_bytes;
+        // unpack signature to unlockblocks
+        let mut unlock_blocks = Vec::new();
+        for _ in 0..input_len {
+            let unlock_block = Unlock::unpack_verified(&mut readable).map_err(|_| crate::Error::PackableError)?;
+            unlock_blocks.push(unlock_block);
+        }
+        Ok(unlock_blocks)
```

Comment:
Remove `blocks`, only called `unlocks` now since we renamed `Message` to `Block`